mig/shared/base.py

```diff
@@ -505,23 +507,28 @@ def force_utf8(val, highlight=''):
         val = "%s" % val
     if not is_unicode(val):
         return val
-    return "%s%s%s" % (highlight, val.encode("utf8"), highlight)
+    return codecs.encode("%s%s%s" % (highlight, val, highlight), 'utf8')
+
+
+def _walk_and_covert_recursive(input_obj, highlight='', _as_bytes=False, _force_primitive=None, _force_recursive=None):
```

Comment:
I guess that's a typo in the function name (convert), right? ... and I'm sure it would like a doc string while at it :)

Reply:
I still get confused about what needs docstrings I guess - in this case its an internal function and both the callers have a docstring. Adding it here would seem to just duplicate what's already there, thats why I didn't add one - absolutely no issue just doing that, just elaborating on why I hadn't already :)

Reply:
Anything non-trivial, really, no matter the internal or public 'scope'. It's as much to help ourselves (re)visiting code not fresh in memory.
A few words like "Shared helper to recursively walk and convert string entries to a certain primitive type" would  suffice here, but better it could e.g. have explained the `_as_bytes` arg, so that the conversion logic gone missing would have been easy to spot during review.